### PR TITLE
Closes #88 Use pharmaversesdtm to host admiralpeds test data

### LINKED ISSUE #88: Feature Request:  Add/Update SDTM test data for {admiralpeds}
### Feature Idea

Add/Update SDTM test data for {admiralpeds}

### Relevant Input

N/A

### Relevant Output

N/A

### Reproducible Example/Pseudo Code

N/A

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
Do we hold off on this until first release of admiralpeds?

**Comment by rossfarrugia** [comment]
> Do we hold off on this until first release of admiralpeds?

Yes, it should be aligned with this update: https://github.com/pharmaverse/admiralpeds/issues/40. I'm just catching up from last week and looks like we had more review comments for admiralpeds and the team are working through them still so no CRAN submission as of yet...

**Comment by rossfarrugia** [comment]
@zdz2101 please let us know if you think this one is now ready for review given we've released admiralpeds to CRAN, and me and @Fanny-Gautier could take a look

**Comment by zdz2101** [comment]
> @zdz2101 please let us know if you think this one is now ready for review given we've released admiralpeds to CRAN, and me and @Fanny-Gautier could take a look

this is ready for review, the one on `admiralpeds` is to demonstrate it does work

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -19,3 +19,6 @@ data-raw/oe_ophtha.R @manciniedoardo
 data-raw/qs_ophtha.R @manciniedoardo
 data-raw/sc_ophtha.R @manciniedoardo
 
+# admiralpeds
+data-raw/dm_peds.R @fanny-gautier
+data_raw/vs_peds.R @fanny-gautier
```
Comment (rossfarrugia) [inline .github/CODEOWNERS:24]:
```suggestion
data-raw/vs_peds.R @fanny-gautier
```

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# pharmaversesdtm 1.1.0
+
```
Comment (rossfarrugia) [inline NEWS.md:2]: Add `## New Features` row

File: R/data.R [hunk 3]
```diff
@@ -316,3 +316,15 @@
 #'
 #' @source Constructed by `{admiralvaccine}` developers
 "suppis_vaccine"
+
+#' Demographic Dataset-pediatrics
+#'
+#' An updated SDTM DM dataset with pediatric patients
+#' @source Constructed by `{admiralpeds}` developers
```
Comment (rossfarrugia) [inline R/data.R:323]: Add blank row above for consistency with the others

File: R/data.R [hunk 4]
```diff
@@ -316,3 +316,15 @@
...
+"dm_peds"
+
+#' Vital signs Dataset-pediatrics
+#'
+#' An updated SDTM VS dataset with anthropometric measurements for pediatric patients
+#' @source Constructed by `{admiralpeds}` developers
```
Comment (rossfarrugia) [inline R/data.R:329]: Add blank row above for consistency with the others

**Comment by rossfarrugia** [comment]
@kaz462 @manciniedoardo @Fanny-Gautier appreciate if you could give this PR a sanity check too.

File: data-raw/vs_peds.R [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    VSSTRESC = as.character(VSSTRESN),
+    VSORRES = VSSTRESC,
+    VSSTRESU = VSORRESU,
+    VSSTAT = NA_character_,
+    VSLOC = NA_character_,
+    VSBLFL = if_else(VISITNUM == 2, "Y", NA_character_),
+    VISITDY = NA_integer_,
+    VSDY = NA_integer_,
+    VSEVAL = NA_character_,
+    EPOCH = "Epoch"
```
Comment (kaz462) [inline data-raw/vs_peds.R:242]: Is it intentional to leave those columns blank (including `VSBLFL`), should we just remove this part and carry values over from the original VS data?

File: data-raw/vs_peds.R [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+      },
+      TRUE ~ VSSTRESN
+    )
+  ) %>%
+  ungroup()
+
+# Formatting the output dataset ----
+vs_peds <- vs_peds %>%
+  mutate(
+    VSSEQ = row_number(),
```
Comment (kaz462) [inline data-raw/vs_peds.R:217]: should `VSSEQ` be derived by `group_by(USUBJID)`? e.g. for each subject, `VSSEQ` starts from 1
image

File: data-raw/vs_peds.R (hunk 5 above)
Comment (bundfussr) [reply data-raw/vs_peds.R:242]: By the way, should `VSBLFL` be replaced by `VSLBOXFL`? In the SDTMIG it is stated that `VSBLFL` is kept only for backward compatibility. See also "4.5.9 Baseline Values" in the SDTMIG.

**Review by Fanny-Gautier — COMMENTED** [review]
minor comments added regarding the use of data()

File: data-raw/vs_peds.R (hunk 6 above)
Comment (zdz2101) [reply data-raw/vs_peds.R:217]: @Fanny-Gautier we had another collaborator make the data here I believe it was @remikk wondering if you or him wanted to handle this fix?

Comment (Fanny-Gautier) [reply data-raw/vs_peds.R]: @zdz2101 this is now fixed

**Comment by zdz2101** [comment]
@rossfarrugia @Fanny-Gautier This can be reviewed once again

File: R/data.R (hunk 3 above)
Comment (rossfarrugia) [reply R/data.R:324]: this was missed so i added it now

**Review by rossfarrugia — APPROVED** [review]
